core/notification/src/main/java/edu/stanford/spezi/core/notification/setting/NotificationSettingViewModel.kt

```diff
@@ -0,0 +1,116 @@
+                }
+            }
+        }
+    }
+
+    fun onAction(action: Action) {
+        when (action) {
+            Action.Back -> {
+                navigator.navigateTo(NavigationEvent.PopBackStack)
+            }
+
+            is Action.SwitchChanged -> {
+                val currentState = _uiState.value
+                if (currentState is UiState.NotificationSettingsLoaded) {
+                    val currentSettings = currentState.notificationSettings
+                    val newSettings =
+                        notificationSettingsMapper.mapSwitchChanged(action, currentSettings)
+                    _uiState.update {
+                        UiState.NotificationSettingsLoaded(
+                            currentSettings.copy(
+                                pendingActions = currentSettings.pendingActions.plus(
+                                    action.notificationType
+                                )
+                            )
+                        )
+                    }
+                    viewModelScope.launch {
+                        repository.saveNotificationSettings(newSettings)
+                            .onFailure {
+                                messageNotifier.notify("Failed to save notification settings")
+                                _uiState.update {
+                                    UiState.NotificationSettingsLoaded(
+                                        currentSettings.copy(
+                                            pendingActions = currentSettings.pendingActions.minus(
+                                                action.notificationType
+                                            )
+                                        )
+                                    )
+                                }
+                            }.onSuccess {
+                                _uiState.update {
+                                    UiState.NotificationSettingsLoaded(
+                                        newSettings.copy(
+                                            pendingActions = currentSettings.pendingActions.minus(
+                                                action.notificationType
+                                            )
+                                        )
+                                    )
+                                }
+                            }
+                    }
+                }
+            }
+        }
+    }
+
+    sealed interface Action {
+        data object Back : Action
+
+        data class SwitchChanged(
+            val notificationType: NotificationType,
+            val isChecked: Boolean,
+        ) : Action
+    }
+
+    sealed interface UiState {
+        data object Loading : UiState
+        data class Error(val message: String) :
+            UiState
+
+        data class NotificationSettingsLoaded(
+            val notificationSettings: NotificationSettings,
+        ) :
+            UiState
```

Comment:
Minor: Would be great to declare the conformance in the same line with `:` here and above, basically just move `UiState` above